apps/client/src/features/app-settings/panel/project-panel/ProjectCreateForm.tsx

```diff
@@ -50,7 +51,12 @@ export default function ProjectCreateForm(props: ProjectCreateFromProps) {
   const handleSubmitCreate = async (values: ProjectCreateFormValues) => {
     try {
       setError(null);
-      const filename = values.title?.trim();
+
+      const maybeFileName = values.title === '' || values.title === undefined ? 'untitled' : values.title.trim();
+
+      const filename = filenamify(`${maybeFileName}.json`, {
+        replacement: '_',
+      });
```

Comment:
I am not really sure about the strategy here, few thoughts:
- we want to avoid adding libraries in the UI side. we can do some rough validation and let the backend deal with it
- why are we accepting falsy title values? I may be missing something, but is there a case where we want to let users not give us a name?

if we are to maintain this logic, since we are not interested in the specifics of the false state, we can simplify the logic to say "if no value is given, we call it untitled", like

```ts
const fileName = !values.title  ? 'untitled' : values.title.trim();
```

Also, since the variable always contains a filename, I wouldnt say it makes sense to call it maybeFileName

Reply:
> why are we accepting falsy title values? I may be missing something, but is there a case where we want to let users not give us a name?

that was the previous implementation where users could leave the Title blank, I'm fine with forcing a file name

We need to do some sort of transformation if we want to allow users to use symbols like `>` in the title